modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -237,363 +196,72 @@ class Instrument_Manager extends \NDB_Menu_Filter
      *
-     * @param string $instrument The instrument name
-     *
-     * @return string The instrument type
+     * @return ?array of site permissions or null
      */
-    protected function checkInstrumentType(string $instrument): string
+    public function allSitePermissionNames() : ?array
     {
-        $linst = $this->_path . "/project/instruments/$instrument.linst";
-        if (file_exists($linst)) {
-            return 'Instrument Builder';
-        }
-
-        $php = $this->_path .
-            "/project/instruments/NDB_BVL_Instrument_$instrument.class.inc";
-
-        if (file_exists($php)) {
-            return 'PHP';
-        }
-
-        return 'Missing';
-    }
-
-    /**
-     * Checks if entries exists in the subtest table for each pages of the
-     * instrument's linst file.
-     *
-     * @param string $instrument The instrument name
-     *
-     * @return string An evocative answer
-     */
-    protected function checkPagesValid(string $instrument): string
-    {
-        $filename = $this->_path . "/project/instruments/$instrument.linst";
-        if (!file_exists($filename)) {
-            return '?';
-        } else {
-            return $this->checkPages($instrument);
-        }
+        return null;
     }
 
     /**
-     * Checks if a table with that instrument name exists
+     * Tells the base class that this page's provisioner can support
+     * the UserProjectMatch filter.
      *
-     * @param string $instrument The instrument name
-     *
-     * @return string An evocative answer
+     * @return bool false
      */
-    protected function checkTableValid(string $instrument): string
+    public function useProjectFilter() : bool
     {
-        $filename = $this->_path . "/project/instruments/$instrument.linst";
-        if (!file_exists($filename)) {
-            return '?';
-        } else {
-            return $this->checkTable($instrument);
-        }
+        return false;
     }
 
     /**
-     * Checks if a table with that instrument name exists
+     * {@inheritDoc}
      *
-     * @param string $instrument The instrument name
-     *
-     * @return string An evocative answer
+     * @return array
      */
-    protected function checkTableInstalled(string $instrument): string
+    protected function getFieldOptions() : array
     {
-        // This should also check that all the columns exist and
-        // have the right type, for new style instruments
-        $exists = $this->_factory->database()->pselectOne(
-            '
-           SELECT count(*)
-           FROM information_schema.tables
-           WHERE TABLE_SCHEMA=:dbname AND
-           TABLE_NAME=:tablename
-          ',
-            [
-                'dbname'    => $this->_factory->settings()->dbName(),
-                'tablename' => $instrument,
-            ]
-        );
-        return ($exists > 0) ? 'Exists' : 'Missing';
-    }
+        $factory = \NDB_Factory::singleton();
+        $db      = $factory->database();
```

Comment:
The database should come from `$this->loris->getDatabaseConnection()`, there's no reason to use the factory singleton.